main/activity-log.js

```diff
@@ -0,0 +1,79 @@
+'use strict'
+
+const Store = require('electron-store')
+const configStore = new Store({
+  name: 'activity-log'
+})
+
+class ActivityLog {
+  #entries
+  #lastId
+
+  constructor () {
+    this.#entries = loadStoredEntries()
+    this.#lastId = this.#entries
+      // We are storing ids as strings to allow us to switch to GUIDs in the future if needed
+      // When looking for the max id used, we need to cast strings to numbers.
+      .map(e => +e.id)
+      // Find the maximum id or return 0 when no events were recorded yet
+      .reduce((max, it) => it > max ? it : max, 0)
+  }
+
+  /**
+   * @param {import('./typings').ActivityEvent} args
+   * @returns {import('./typings').ActivityEntry}
```

Comment:
would it be possible to import these types in the file header, or is this a limitation of our current type setup? I find these lines quite noisy

Reply:
I agree with you this is noisy.

You can import types, but it's a bit verbose, so I usually take the lazy approach you can see here. I will change my code to import those types so you can see which version you prefer.

BTW this DX issue has been discussed in  https://github.com/microsoft/TypeScript/issues/41825. Unfortunately, there was no solution yet.

Reply:
See [8a759de](https://github.com/filecoin-project/filecoin-station/pull/110/commits/8a759de1fdff4643442bf8e258ba49d0914d2789)